crates/oxc_linter/src/rules/eslint/func_names.rs

```diff
@@ -0,0 +1,507 @@
+use oxc_ast::ast::{
+    AssignmentTarget, AssignmentTargetProperty, BindingPatternKind, Expression, Function,
+    FunctionType, PropertyKind,
+};
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{Atom, Span};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Default, Clone)]
+pub struct FuncNames {
+    default_config: FuncNamesConfig,
+    generators_config: FuncNamesConfig,
+}
+
+#[derive(Debug, Default, Clone, PartialEq)]
+enum FuncNamesConfig {
+    #[default]
+    Always,
+    AsNeeded,
```

Comment:
I feel like this would make a more useful default, what do you think?

Reply:
From the offical docs:

>This rule has a string option:
> -  "always" (default) requires function expressions to have a name
> - "as-needed" requires function expressions to have a name, if the name isn’t assigned automatically per the ECMAScript specification.
> - "never" disallows named function expressions, except in recursive functions, where a name is needed

Reply:
I agree that this is the original implementation of the rule, but some of their defaults are rather strict and often get changed. I'm thinking of, for example, `eqeqeq`, where `smart` would make a better default. Do you have an opinion on actual usability?

Reply:
I don't have an opinion which options should be the default one. But I would prefer to be the same as the eslint one. 

Someone who has in the config just `"eslint/func-names": "error"` will get another result with oxlint then with eslint.

If I need to choose, I would choose `never` :')